packages/providers/src/core/PoapMomentsApi/PoapMomentsApi.ts

```diff
@@ -120,37 +130,26 @@ export class PoapMomentsApi implements MomentsApiProvider {
    * @param {CreateMomentInput} input - The input for creating a moment
    * @returns {Promise<CreateMomentResponse>} - A Promise that resolves to the created moment
    */
-  async createMoment(input: CreateMomentInput): Promise<CreateMomentResponse> {
-    return await this.secureFetch(`${this.baseUrl}/moments`, {
-      method: 'POST',
-      body: JSON.stringify(input),
-      headers: { 'Content-Type': 'application/json' },
+  public async createMoment(
+    input: CreateMomentInput,
+  ): Promise<CreateMomentResponse> {
+    const response = await axios.post(`${this.baseUrl}/moments`, input, {
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: await this.getAuthorizationToken(),
+      },
     });
-  }
 
-  /**
-   * Sends a secure HTTP request to the Poap Registry API.
-   *
-   * @async
-   * @private
-   * @function
-   * @name PoapRegistryApi#secureFetch
-   * @param {string} url - The URL for the HTTP request.
-   * @param {any} options - The options for the HTTP request.
-   * @returns {Promise<any>} A Promise that resolves with the response from the API.
-   */
-  async secureFetch(url: string, options: any): Promise<any> {
-    const headersWithApiKey = {
-      ...options.headers,
-      'x-api-key': this.apiKey,
-    };
+    return response.data;
+  }
 
-    const response = await axios(url, {
-      method: options.method,
-      data: options.body,
-      headers: headersWithApiKey,
-    });
+  private async getAuthorizationToken(): Promise<string> {
+    if (!this.authenticationProvider) {
+      throw new Error(
```

Comment:
I understand why is optional but the error at this stage it make noice that and block the dev in the aftermath. If is better to keep it optional and make a whole for read Nf write, would be nicer to make the error throw before and let the dev know before hand.